src/js/closable-announcements.js

```diff
@@ -29,9 +29,41 @@
 
       function closeAnnouncement(announcement) {
         addToAnnouncementStorage(announcement);
-        announcement.style = `--js-announcement-height: ${announcement.offsetHeight}px`; // Set fixed height to allow CSS animation
+        // Set fixed height to allow CSS animation
+        announcement.style = `--js-announcement-height: ${announcement.offsetHeight}px`;
         window.setTimeout(() => {
+          let focused = false;
+          const allAnnouncements = document.querySelectorAll('.announcement:not(.js-announcement--hide)');
           announcement.classList.add(ANNOUCEMENT_HIDE);
+          // Moving focus to correct place after closing the announcement.
+          // NOTICE: This is very dependent on the html-structure of the header and its components.
+          // Go through all announcements and use the uuid to determine which focusable announcement is before
+          // the closed announcement.
+          for (let i = 0, max = allAnnouncements.length; i < max; i++) {
+            if (allAnnouncements[i].dataset.uuid === announcement.dataset.uuid && i > 0) {
+              const focusableAnnouncement = allAnnouncements[i - 1].querySelectorAll('.announcement__close, .announcement__link a');
+              focusableAnnouncement[focusableAnnouncement.length - 1].focus();
+              focused = true;
+              break;
+            }
+          }
+          // If there is no more announcements to focus to we need to figure out if we are viewing the page with mobile
+          // or desktop navigation. We check if the header-bottom menu is visible to determine this.
+          if (focused === false) {
+            const desktopMenu = document.querySelectorAll('.header-bottom .desktop-menu > .menu');
+            for (let i = 0, max = desktopMenu.length; i < max; i++) {
+              const desktopMenuHidden = window.getComputedStyle(desktopMenu[i]).display === 'none';
+              // Depending on the visible menu we move the focus straight to nav-toggle buttons or the menu links if no
+              // breadcrumb is set.
+              if (desktopMenuHidden === true) {
+                const focusableElements = document.querySelectorAll('.nav-toggle__button, .breadcrumb a:last-of-type');
+                focusableElements[focusableElements.length - 1].focus();
+              } else {
```

Comment:
And another comment about this being the desktop mode

Reply:
Comment added.